### PR TITLE
Update to Rust 1.0.36

### PR DESCRIPTION
**Pull request by jedisct1**
Remove now deprecated `std::mem::uninitialized()` usage.

`rustfmt` includes bug fixes but doesn't seem to change the indentation rules.

### CODE REVIEW

**Comment by awortman-fastly** [comment]
:tada: down with `mem::uninitialized`! - small request, could you reword the commit (and PR) to `Rust 1.36.0`, please :)

**Comment by jedisct1** [comment]
Haha, of course :) Done!

File: lucet-runtime/lucet-runtime-internals/src/instance/signals.rs [hunk 1]
```diff
@@ -374,15 +375,15 @@ pub struct SigStack {
 
 impl SigStack {
     pub fn new(sp: *mut libc::c_void, flags: SigStackFlags, size: libc::size_t) -> SigStack {
-        let mut stack = unsafe { std::mem::uninitialized::<libc::stack_t>() };
+        let mut stack: libc::stack_t = unsafe { mem::uninitialized() };
```
Comment (acfoltzer) [inline lucet-runtime/lucet-runtime-internals/src/instance/signals.rs]: We explicitly initialize all the fields of `libc::stack_t` here, so we don't actually need to use uninitialized memory at all.

File: lucet-runtime/lucet-runtime-internals/src/instance/signals.rs [hunk 2]
```diff
@@ -374,15 +375,15 @@ pub struct SigStack {
...
         stack.ss_sp = sp;
         stack.ss_flags = flags.bits();
         stack.ss_size = size;
         SigStack { stack }
     }
 
     pub fn disabled() -> SigStack {
-        let mut stack = unsafe { std::mem::uninitialized::<libc::stack_t>() };
+        let mut stack: libc::stack_t = unsafe { mem::uninitialized() };
```
Comment (acfoltzer) [inline lucet-runtime/lucet-runtime-internals/src/instance/signals.rs]: same as above

File: lucet-runtime/lucet-runtime-internals/src/instance/signals.rs [hunk 3]
```diff
@@ -437,11 +438,8 @@ pub unsafe fn sigaltstack(new_sigstack: Option<SigStack>) -> nix::Result<Option<
 }
 
 pub unsafe fn altstack_flags() -> nix::Result<SigStackFlags> {
-    let mut current_stack = std::mem::uninitialized::<libc::stack_t>();
-    let res = libc::sigaltstack(
-        std::ptr::null_mut(),
-        &mut current_stack as *mut libc::stack_t,
-    );
+    let mut current_stack = MaybeUninit::<libc::stack_t>::uninit();
+    let res = libc::sigaltstack(std::ptr::null_mut(), current_stack.as_mut_ptr());
```
Comment (acfoltzer) [inline lucet-runtime/lucet-runtime-internals/src/instance/signals.rs]: need to call `current_stack.assume_init()` after this line

File: lucet-runtime/lucet-runtime-internals/src/instance/signals.rs [hunk 4]
```diff
@@ -374,15 +375,15 @@ pub struct SigStack {
 
 impl SigStack {
     pub fn new(sp: *mut libc::c_void, flags: SigStackFlags, size: libc::size_t) -> SigStack {
-        let mut stack = unsafe { std::mem::uninitialized::<libc::stack_t>() };
+        let mut stack = unsafe { mem::uninitialized::<libc::stack_t>() };
         stack.ss_sp = sp;
```
Comment (acfoltzer) [inline lucet-runtime/lucet-runtime-internals/src/instance/signals.rs]: We explicitly initialize all the fields of `libc::stack_t` here, so we don't actually need to use uninitialized memory at all.

File: lucet-runtime/lucet-runtime-internals/src/instance/signals.rs [hunk 5]
```diff
@@ -374,15 +375,15 @@ pub struct SigStack {
...
         stack.ss_flags = flags.bits();
         stack.ss_size = size;
         SigStack { stack }
     }
 
     pub fn disabled() -> SigStack {
-        let mut stack = unsafe { std::mem::uninitialized::<libc::stack_t>() };
+        let mut stack = unsafe { mem::uninitialized::<libc::stack_t>() };
         stack.ss_sp = std::ptr::null_mut();
```
Comment (acfoltzer) [inline lucet-runtime/lucet-runtime-internals/src/instance/signals.rs]: same as above

File: lucet-runtime/lucet-runtime-internals/src/instance/signals.rs (hunk 4 above)
Comment (jedisct1) [reply lucet-runtime/lucet-runtime-internals/src/instance/signals.rs]: Right. The last `mem::uninitialized` bits have been removed 🎉

**Pull request merged**